src/session_impl.cpp

```diff
@@ -6599,10 +6614,19 @@ namespace {
 		if (m_natpmp) m_natpmp->delete_mapping(handle);
 	}
 
+	void session_impl::stop_ip_notifier()
+	{
+		if (!m_ip_notifier) return;
+
+		m_ip_notifier->cancel();
+		m_ip_notifier.reset();
```

Comment:
This means the notifier object can be deleted while there is a notification event is queued. It doesn't look like this poses a problem for any of the notifier implementations, but the netlink version should be changed to make `ip_change_notifier_impl::on_notify` a static function because `this` is not used and with this change may be a dangling pointer.

Reply:
isn't the handler object itself held by the notifier object?

Reply:
No, it's captured in the `std::bind` object.

Reply:
thanks for the review, I will accommodate the code for it